test/unit/associations/has-many.test.js

```diff
@@ -212,6 +212,85 @@ describe(Support.getTestDialectTeaser('hasMany'), () => {
       }
     });
   });
+
+  describe('set', async () => {
+    it('should unassociate (delete) addr1 given a non-null constraint on userId', async () => {
+      const User = current.define('User', {
+        username: DataTypes.STRING,
+        UserId: {
+          type: DataTypes.INTEGER,
+          allowNull: false,
+        },
+      });
+
+      const Address = current.define('Address', {
+        street: DataTypes.STRING,
+      });
+
+      Address.belongsTo(User, {
+        foreignKey: {
+          allowNull: false,
+        },
+      });
+      User.hasMany(Address);
+      await current.sync();
+
+      const jane = await User.create({
+        username: 'jane',
+        UserId: 1,
+      });
+
+      const addr1 = await jane.createAddress({ street: 'st' });
+      const addr2 = await Address.build({
+        street: 'st2',
+        UserId: jane.id,
+      });
+      await addr2.save();
```

Comment:
build isn't async, `create` is (create = build+save)

```suggestion
      const addr2 = await Address.create({
        street: 'st2',
        UserId: jane.id,
      });
```

Can you also put these inside a Promise.all to do them in parallel?

```typescript
      const [addr1, addr2] = await Promise.all([
        jane.createAddress({ street: 'st' }),
        Address.create({
          street: 'st2',
          UserId: jane.id,
        }),
      ]);
```